packages/grpc-js/src/load-balancer-cds.ts

```diff
@@ -0,0 +1,162 @@
+ * limitations under the License.
+ *
+ */
+
+import {
+  LoadBalancer,
+  ChannelControlHelper,
+  registerLoadBalancerType,
+} from './load-balancer';
+import { SubchannelAddress } from './subchannel';
+import {
+  LoadBalancingConfig,
+  isCdsLoadBalancingConfig,
+  EdsLbConfig,
+  CdsLoadBalancingConfig,
+} from './load-balancing-config';
+import { XdsClient, Watcher } from './xds-client';
+import { ChildLoadBalancerHandler } from './load-balancer-child-handler';
+import { Cluster__Output } from './generated/envoy/api/v2/Cluster';
+import { ConnectivityState } from './channel';
+import { UnavailablePicker } from './picker';
+import { Status } from './constants';
+import { Metadata } from '.';
+
+const TYPE_NAME = 'cds';
+
+export class CdsLoadBalancer implements LoadBalancer {
+  private childBalancer: ChildLoadBalancerHandler;
+  private xdsClient: XdsClient | null = null;
+  private watcher: Watcher<Cluster__Output>;
+
+  private isWatcherActive = false;
+
+  private latestCdsUpdate: Cluster__Output | null = null;
+
+  private latestConfig: CdsLoadBalancingConfig | null = null;
+  private latestAttributes: { [key: string]: unknown } = {};
+
+  constructor(private readonly channelControlHelper: ChannelControlHelper) {
+    this.childBalancer = new ChildLoadBalancerHandler(channelControlHelper);
+    this.watcher = {
+      onValidUpdate: (update) => {
+        this.latestCdsUpdate = update;
+        const edsConfig: EdsLbConfig = {
+          cluster: update.name,
+          edsServiceName:
+            update.eds_cluster_config!.service_name === ''
+              ? undefined
+              : update.eds_cluster_config!.service_name,
+          localityPickingPolicy: [],
+          endpointPickingPolicy: [],
+        };
+        if (update.lrs_server?.self) {
+          /* the lrs_server.self field indicates that the same server should be
+           * used for load reporting as for other xDS operations. Setting
+           * lrsLoadReportingServerName to the empty string sets that behavior.
+           * Otherwise, if the field is omitted, load reporting is disabled. */
+          edsConfig.lrsLoadReportingServerName = '';
+        }
+        this.childBalancer.updateAddressList(
+          [],
+          { name: 'eds', eds: edsConfig },
+          this.latestAttributes
+        );
+      },
+      onResourceDoesNotExist: () => {
+        this.xdsClient?.removeClusterWatcher(
+          this.latestConfig!.cds.cluster,
+          this.watcher
+        );
+        this.isWatcherActive = false;
+      },
+      onTransientError: (status) => {
+        if (this.latestCdsUpdate === null) {
+          channelControlHelper.updateState(
+            ConnectivityState.TRANSIENT_FAILURE,
+            new UnavailablePicker({
+              code: Status.UNAVAILABLE,
+              details: `xDS request failed with error ${status.details}`,
+              metadata: new Metadata(),
+            })
+          );
+        }
+      },
+    };
+  }
+
```

Comment:
So there is no need for a `updateChild()` function here?

Reply:
Since the only thing in the CDS config is the name, a config change always invalidates the previous update, so that update would never do anything.